src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminal.accessibility.contribution.ts

```diff
@@ -48,56 +53,219 @@ class TextAreaSyncContribution extends DisposableStore implements ITerminalContr
+		const capability = this._instance.capabilities.get(TerminalCapability.CommandDetection);
+		const commands = capability?.commands;
+		const currentCommand = capability?.currentCommand;
+		if (!commands?.length) {
+			return;
+		}
+		const result: ICommandWithEditorLine[] = [];
+		for (const command of commands) {
+			const lineNumber = this._getEditorLineForCommand(command);
+			if (!lineNumber) {
+				continue;
+			}
+			result.push({ command, lineNumber });
+		}
+		if (currentCommand) {
+			const lineNumber = this._getEditorLineForCommand(currentCommand);
+			if (!!lineNumber) {
+				result.push({ command: currentCommand, lineNumber });
+			}
+		}
+		return result;
+	}
+	private _getEditorLineForCommand(command: ITerminalCommand | ICurrentPartialCommand): number | undefined {
+		let line: number | undefined;
+		if ('marker' in command) {
+			line = command.marker?.line;
+		} else if ('commandStartMarker' in command) {
+			line = command.commandStartMarker?.line;
+		}
+		if (line === undefined || line < 0) {
+			return;
+		}
+		line = this._bufferTracker.bufferToEditorLineMapping.get(line);
+		if (line === undefined) {
+			return;
+		}
+		return line + 1;
 	}
+}
+interface ICommandWithEditorLine { command: ITerminalCommand | ICurrentPartialCommand; lineNumber: number }
+
 
+export class TerminalAccessibleViewContribution extends Disposable implements ITerminalContribution {
+	static readonly ID = 'terminal.accessibleBufferProvider';
+	static get(instance: ITerminalInstance): TerminalAccessibleViewContribution | null {
+		return instance.getContribution<TerminalAccessibleViewContribution>(TerminalAccessibleViewContribution.ID);
+	}
+	private _bufferTracker: BufferContentTracker | undefined;
+	private _xterm: Pick<IXtermTerminal, 'shellIntegration' | 'getFont'> & { raw: Terminal } | undefined;
+	constructor(
+		private readonly _instance: ITerminalInstance,
+		processManager: ITerminalProcessManager,
+		widgetManager: TerminalWidgetManager,
+		@IAccessibleViewService private readonly _accessibleViewService: IAccessibleViewService,
+		@IInstantiationService private readonly _instantiationService: IInstantiationService,
+		@ITerminalService private readonly _terminalService: ITerminalService) {
+		super();
+		this._register(AccessibleViewAction.addImplementation(90, 'terminal', () => {
+			if (this._terminalService.activeInstance !== this._instance) {
+				return false;
+			}
+			this.show();
+			return true;
+		}, TerminalContextKeys.focus));
+	}
+	xtermReady(xterm: IXtermTerminal & { raw: Terminal }): void {
+		const addon = this._instantiationService.createInstance(TextAreaSyncAddon, this._instance.capabilities);
+		xterm.raw.loadAddon(addon);
+		addon.activate(xterm.raw);
+		this._xterm = xterm;
+	}
+	show(): void {
+		if (!this._xterm) {
+			return;
+		}
+		if (!this._bufferTracker) {
+			this._bufferTracker = this._instantiationService.createInstance(BufferContentTracker, this._xterm);
+		}
```

Comment:
`BufferContentTracker` creates disposables (eg. `_lastCachedMarker`) but is not disposable.